src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs

```diff
@@ -196,6 +199,31 @@ protected virtual Task SetResultTagsAsync(Activity activity, ExecutionOptions ex
         return Task.CompletedTask;
     }
 
+    private bool Filter(ExecutionOptions options, Activity activity)
+    {
+        try
+        {
+            if (!_telemetryOptions.Filter(options))
+            {
+                activity.IsAllDataRequested = false;
+                activity.ActivityTraceFlags &= ~ActivityTraceFlags.Recorded;
+                return false;
+            }
+        }
+        catch (Exception ex)
+        {
+#if NET6_0_OR_GREATER
+            activity.SetStatus(ActivityStatusCode.Error);
+#else
+            activity.SetTag("otel.status_code", "ERROR");
+#endif
+            _telemetryOptions.EnrichWithException(activity, ex);
```

Comment:
Many instrumentation libraries provide `RecordException` boolean to [record the exception information](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/exceptions/exceptions-spans.md) on the span. `OpenTelemetry.Api` provides a convenient extension method to implement this functionality. I'd like to add this option, too, but since we don't reference the `OpenTelemetry.Api` package, we will have to copy their code. Strictly saying, this is not really necessary because the user can do that in `EnrichWithException` delegate. Boolean is just more convenient. What do you think?

Reply:
We could create a protected method with default functionality.  It seems to me that the default delegate in the options class should do nothing, and only be used for adding additional tags.